src/community/security/keycloak/src/main/java/org/geoserver/security/keycloak/AuthResults.java

```diff
@@ -49,16 +49,24 @@ public AuthResults(AuthChallenge challenge) {
      * @param authentication valid credentials
      */
     public AuthResults(Authentication authentication) {
-        assert authentication.getDetails() instanceof SimpleKeycloakAccount;
-        final SimpleKeycloakAccount details = (SimpleKeycloakAccount) authentication.getDetails();
+        Object username = null;
+        Object details = null;
+        if (authentication.getDetails() instanceof SimpleKeycloakAccount) {
```

Comment:
Is there a case where we are using a keycloak plugin and not getting a SimpleKeycloakAccount?

Reply:
Yeps, when a user has logged in through the form method and the security context is already available. The plugin checks first if there's a principal already present into the security context with a matching username. In that case it brings the roles from the latter.